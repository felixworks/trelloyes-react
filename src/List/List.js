import React from 'react';
import Card from '../Card/Card.js';
import './List.css';

function List(props) {
    console.log('list props', props);
    console.log('list of cardIds', props.cardIds);
    return (
        <section className="List">
            <header className="List-header">
                <h2>{props.header}</h2>
            </header>
            {props.cardIds.map((cardId, i) => <Card 
            cardId={cardId} 
            key={i}
            title={props.allCards[cardId].title}
            content={props.allCards[cardId].content}
            />)}
            
        </section>
    )
}

export default List;