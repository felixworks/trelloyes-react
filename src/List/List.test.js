import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List.js';
import STORE from '.././store.js';

describe('List test', () => {
    it('renders without crashing', () => {
        const div = document.createElement("div");
        ReactDOM.render(<List 
            header={STORE.lists[0].header} 
            key={STORE.lists[0].id} 
            cardIds={STORE.lists[0].cardIds} 
            allCards={STORE.allCards}/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders List UI as expected', () => {
        const tree = renderer
            .create(<List 
                header={STORE.lists[0].header} 
                key={STORE.lists[0].id} 
                cardIds={STORE.lists[0].cardIds} 
                allCards={STORE.allCards} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});