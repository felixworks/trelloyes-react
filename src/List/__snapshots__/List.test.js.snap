// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List test renders List UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      First list
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        Second card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        Fifth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        Sixth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        Seventh card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        Tenth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        Twelth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        Thirteenth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
</section>
`;
